query/src/sql/planner/binder/ddl/table.rs

```diff
@@ -369,8 +372,32 @@ impl<'a> Binder {
                     .collect();
                 (DataSchemaRefExt::create(fields), vec![])
             }
-            // TODO(leiysky): Support `CREATE TABLE AS SELECT` with specified column definitions
-            _ => Err(ErrorCode::UnImplement("Unsupported CREATE TABLE statement"))?,
+            (Some(source), Some(query)) => {
+                // e.g. `CREATE TABLE t (i INT) AS SELECT * from old_t` with columns speicified
+                let (source_schema, source_coments) =
+                    self.analyze_create_table_schema(source).await?;
+                let init_bind_context = BindContext::new();
+                let (_s_expr, bind_context) = self.bind_query(&init_bind_context, query).await?;
+                let query_fields: Vec<DataField> = bind_context
+                    .columns
+                    .iter()
+                    .map(|column_binding| {
+                        DataField::new(
+                            &column_binding.column_name,
+                            *column_binding.data_type.clone(),
+                        )
+                    })
+                    .collect();
+                let  source_fields = source_schema.fields().clone();
```

Comment:
I'm not sure but does this need a reformat?

Reply:
Weird, check passed...

Reply:
I think it's a bug of `rustfmt`.

![image](https://user-images.githubusercontent.com/5351546/179002435-1f9229ec-d6b5-4e86-a1de-db99e1dbf636.png)

Reply:
Code in this block won't be formatted, for example:

![image](https://user-images.githubusercontent.com/5351546/179006056-d1e5fd0c-c5b2-4a2f-9fc6-4686304dfaea.png)

Reply:
This format bug will be addressed by https://github.com/datafuselabs/databend/pull/6631